base/linalg/generic.jl

```diff
@@ -795,6 +795,19 @@ function inv(A::AbstractMatrix{T}) where T
     A_ldiv_B!(factorize(convert(AbstractMatrix{S}, A)), eye(S0, checksquare(A)))
 end
 
+function pinv(v::AbstractVector{T}, tol::Real=real(zero(T))) where T
```

Comment:
Hm, I was under the impression that `pinv(::StridedMatrix{T}, tol)` ignores all singular values below `tol`, but the threshold is actually `tol * maximum(singular_values)`, so for the vector case, it only matters whether `tol < 1`. So the with the computed default `tol` consistent with the matrix case, an all-zero vector would be returned if the input is all-zero, which makes sense, or if `length(v) > 1/eps(real(float(one(T))))`, which strikes me as pretty bizarre.

I think I'd rather keep the `tol=0` default than aim for maximum consistency here. Opinions?